integration-core/src/main/java/org/symphonyoss/integration/core/bridge/StreamServiceImpl.java

```diff
@@ -37,44 +43,58 @@
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.Collections;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 
 import javax.annotation.PostConstruct;
 
 /**
+ * Service component responsible to post message through the Agent Message API.
  * Created by rsanchez on 13/05/16.
  */
 @Component
 public class StreamServiceImpl implements StreamService {
 
   private static final Logger LOG = LoggerFactory.getLogger(StreamServiceImpl.class);
 
+  private static final String AGENT_SERVICE_NAME = "Agent";
+
+  private static final String AGENT_API_V3 = "1.46.0";
+
   @Autowired
   private AuthenticationProxy authenticationProxy;
 
   @Autowired
-  private AgentApiClient agentApiClient;
+  private AgentApiClient agentV2ApiClient;
+
+  @Autowired
+  private AgentApiClient agentV3ApiClient;
 
   @Autowired
   private PodHttpApiClient podApiClient;
 
   /**
-   * Agent Message API Client
+   * Pod Stream API Client
    */
-  private MessageApiClient messagesApi;
+  private StreamApiClient streamsApi;
 
   /**
-   * Pod Stream API Client
+   * Select the correct Agent Message API Client according to the MessageML version
    */
-  private StreamApiClient streamsApi;
+  private Map<MessageMLVersion, MessageApiClient> apiResolver = new HashMap<>();
 
   /**
    * Initialize
    */
   @PostConstruct
   public void init() {
-    messagesApi = new MessageApiClient(agentApiClient);
     streamsApi = new StreamApiClient(podApiClient);
+
+    MessageApiClient messageApiClient = new V2MessageApiClient(agentV2ApiClient);
+
+    apiResolver.put(MessageMLVersion.V1, messageApiClient);
```

Comment:
Same for both? 

Ok.  I see.  It gets overridden below when version updated event is handled.  You should add some comments to explain that.  At first glance, this looks like a bug.